cmd/pbm/restore.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"log"
+
+	"github.com/pkg/errors"
+
+	"github.com/percona/percona-backup-mongodb/pbm"
+)
+
+func restore(cn *pbm.PBM, bcpName string) error {
+	locks, err := cn.GetLocks(&pbm.LockHeader{})
+	if err != nil {
+		log.Println("get locks", err)
+	}
+
+	ts, err := cn.ClusterTime()
+	if err != nil {
+		return errors.Wrap(err, "read cluster time")
+	}
+
+	// Stop if there is some live operation.
+	// But if there is some stale lock leave it for agents to deal with.
+	for _, l := range locks {
+		if l.Heartbeat.T+pbm.StaleFrameSec >= ts.T {
+			return errors.Errorf("another operation in progress, %s/%s", l.Type, l.BackupName)
+		}
+	}
+
+	bcp, err := cn.GetBackupMeta(bcpName)
```

Comment:
I think it is better to check backup first and lock exactly before SendCmd - in such case it is harder to get race condition from different pbmctl

Reply:
What exactly scenario do you mean by the race condition in from a different pbmctl?

Reply:
I mean when we run restore command in parallel in two terminals exactly in the same second